ui/apps/platform/src/Containers/Clusters/ClustersTable.tsx

```diff
@@ -0,0 +1,246 @@
+import ClusterStatus from './Components/ClusterStatus';
+import CredentialExpiration from './Components/CredentialExpiration';
+import HelmIndicator from './Components/HelmIndicator';
+import OperatorIndicator from './Components/OperatorIndicator';
+import SensorUpgrade from './Components/SensorUpgrade';
+
+export type ClustersTableProps = {
+    clusterIdToRetentionInfo: ClusterIdToRetentionInfo;
+    tableState: TableUIState<Cluster>;
+    selectedClusterIds: string[];
+    onClearFilters: () => void;
+    onDeleteCluster: (cluster: Cluster) => (event: React.MouseEvent) => void;
+    toggleAllClusters: () => void;
+    toggleCluster: (clusterId) => void;
+};
+
+export const COL = {
+    STATUS: 'status',
+    SENSOR: 'sensor',
+} as const;
+type ColId = (typeof COL)[keyof typeof COL];
+
+type ExpansionMap = Record<string, ColId | null>;
+
+function ClustersTable({
+    clusterIdToRetentionInfo,
+    tableState,
+    selectedClusterIds,
+    onClearFilters,
+    onDeleteCluster,
+    toggleAllClusters,
+    toggleCluster,
+}: ClustersTableProps) {
+    const metadata = useMetadata();
+    const [expanded, setExpanded] = useState<ExpansionMap>({});
+
+    function toggle(clusterId: string, col: ColId) {
+        setExpanded((prev) => ({
+            ...prev,
+            [clusterId]: prev[clusterId] === col ? null : col,
+        }));
+    }
+
+    function isCellExpanded(clusterId: string, col: ColId) {
+        return expanded[clusterId] === col;
+    }
+
+    function isRowExpanded(clusterId: string) {
+        return expanded[clusterId] != null;
+    }
+
+    function isHelmManaged(cluster: Cluster) {
+        return (
+            cluster.managedBy === 'MANAGER_TYPE_HELM_CHART' ||
+            (cluster.managedBy === 'MANAGER_TYPE_UNKNOWN' && !!cluster.helmConfig)
+        );
+    }
+
+    function isOperatorManaged(cluster: Cluster) {
+        return cluster.managedBy === 'MANAGER_TYPE_KUBERNETES_OPERATOR';
+    }
+
+    const colSpan = 9;
+
+    return (
+        <Table>
+            <Thead>
+                <Tr>
+                    <Th
+                        select={{
+                            onSelect: () => toggleAllClusters(),
+                            isSelected:
+                                tableState.type === 'COMPLETE' &&
+                                tableState.data.length === selectedClusterIds.length,
+                        }}
+                    />
+                    <Th>Cluster</Th>
+                    <Th>Provider (Region)</Th>
+                    <Th>Status</Th>
+                    <Th>Sensor upgrade status</Th>
+                    <Th>Credential expiration</Th>
+                    <Th>Cluster deletion</Th>
+                </Tr>
+            </Thead>
+            <TbodyUnified
+                tableState={tableState}
+                colSpan={colSpan}
+                errorProps={{
+                    title: 'There was an error loading cluster information',
+                }}
```

Comment:
What should we display when there is no data? I think we can utilize the `emptyProps` prop here.

Reply:
My reasoning for not adding was that `ClustersTablePanel` will render a `NoClustersPage` component, however that could always change so I'll add `emptyProps`.